massa-pool-worker/src/operation_pool.rs

```diff
@@ -84,18 +84,25 @@ impl OperationPool {
         let max_slot = max(max_slot, min_slot);
 
         // search for all our PoS draws in the interval of interest
-        let mut pos_draws = BTreeSet::new();
-        let addrs = self.wallet.read().keys.keys().copied().collect::<Vec<_>>();
-        for addr in addrs {
-            let (d, _) = self
-                .channels
-                .selector
-                .get_address_selections(&addr, min_slot, max_slot)
-                .expect("could not get PoS draws");
-            pos_draws.extend(d.into_iter());
-        }
+        let addrs: PreHashSet<Address> = self.wallet.read().keys.keys().copied().collect();
+        let mut pos_draws: BTreeSet<Slot> = self
+            .channels
+            .selector
+            .get_available_selections_in_range(min_slot..=max_slot, Some(&addrs))
```

Comment:
Can't we return an iterator here (as we need it right after it)?

Reply:
cc @damip

Reply:
We can't return an iterator because it would iterate on a locally owned variable. It's like returning a reference to a a function-local object.

However we could improve things a bit by including the filtering condition of the subsequent `retain` directly within the `filter_map` to avoid a second full iteration

Reply:
Unless you mean that the method `get_available_selections_in_range` itself returns an iterator but this would require keeping a lock in the selector for as long as the iterator is in use so it's not optimal either

Reply:
See #4282